test/Marketplace.test.js

```diff
@@ -478,7 +489,7 @@ describe("Marketplace", function () {
 
       const endBalance = await token.balanceOf(host.address)
       const expectedPartialPayout =
-        (request.expiry - fillTimestamp) * request.ask.reward
+        (request.expiry - fillSeconds - 1) * request.ask.reward // TODO: I don't understand why there needs to be -1 offset
```

Comment:
Found it!
You are right about some mined transaction that advances the clock but it is not of `freeSlot` call. It does not matter at what block's timestamp you are calling the `freeSlot` as the payout is always for the interval `<filledAt;expiryAt>`.

It was because of the `approve()` call in the `beforeEach` statement that advanced the clock after `requestStorage`. TBH the way how I am calculating the partial payout is not really that "fool-proof" 😅